assets/src/edit-story/app/prepublish/warning/accessibility.js

```diff
@@ -49,6 +58,236 @@ function getSpansFromContent(content) {
   );
 }
 
+function getOverlappingArea(a, b) {
+  const dx = Math.min(a.endX, b.endX) - Math.max(a.startX, b.startX);
+  const dy = Math.min(a.endY, b.endY) - Math.max(a.startY, b.startY);
+  if (dx >= 0 && dy >= 0) {
+    return dx * dy;
+  }
+  return 0;
+}
+
+function isOverlapSignificant(overlapArea, textBoxArea) {
+  return overlapArea >= 0.5 * textBoxArea;
+}
+
+function getMostOverlapForElement(
+  textElement,
+  potentialBackgroundElements,
+  pageSize
+) {
+  const textPos = getBox(textElement, pageSize?.width, pageSize?.height);
+  const textBox = getBoundRect([textPos]);
+  const textBoxArea = textBox.width * textBox.height;
+
+  const elementOverlaps = [...potentialBackgroundElements]
+    .filter(({ isBackground }) => !isBackground)
+    .map((element, index) => {
+      const bgPos = getBox(element, pageSize?.width, pageSize?.height);
+      const bgBox = getBoundRect([bgPos]);
+      const overlappingArea = getOverlappingArea(textBox, bgBox);
```

Comment:
This is great but I wonder if we should account for padding in text boxes. This can shift metrics for what is overlapping, but I would suggest we account for that as an iterative improvement—not necessarily in this PR.

Reply:
yeah, I wanna be able to do stuff like this:
![Screen Shot 2020-12-16 at 3 04 03 PM](https://user-images.githubusercontent.com/41136059/102411892-f3f9e780-3faf-11eb-80b0-5d206112dcf4.png)

..without being warned about contrast. Using ~50% overlap was a too high per Will's feedback (only one element will be checked if we go for 50% overlap) but going to 25-30% was a little too low for some of my experiments which I felt were reasonable amounts of "low contrast". I'm gonna go with a middle ground (33%) and see how that feels to everyone. After playing with it a bit, this would be a really good addition to avoid false positives. We have some work coming up around textbox padding, too, so I agree about the iterative improvement.